edx_proctoring/api.py

```diff
@@ -494,6 +496,93 @@ def remove_allowance_for_user(exam_id, user_id, key):
         emit_event(exam, 'allowance.deleted', override_data=data)
 
 
+def add_bulk_allowances(exam_ids, user_ids, allowance_type, value):
+    """
+    Adds (or updates) an allowance for multiple users and exams
+    """
+    exam_ids = set(exam_ids)
+    user_ids = set(user_ids)
+
+    log_message = (
+        'Adding allowances of type allowance_type={allowance_type} with value={value} '
+        'for exams exam_ids={exam_ids} and users user_ids={user_ids}'.format(
+            allowance_type=allowance_type,
+            value=value,
+            exam_ids=exam_ids,
+            user_ids=user_ids
+        )
+    )
+    log.info(log_message)
+    if allowance_type == constants.TIME_MULTIPLIER:
+        err_msg = (
+            u'allowance_value "{value}" should be a float value greater than 1.'
+        ).format(value=value)
+        try:
+            multiplier = float(value) - 1
+        except ValueError:
+            raise AllowanceValueNotAllowedException(err_msg)
+        if multiplier >= 0:
+            for exam_id in exam_ids:
+                try:
+                    target_exam = get_exam_by_id(exam_id)
+                except ProctoredExamNotFoundException:
+                    log_message = (
+                        'Attempted to get exam_id={exam_id}, but this exam does not exist.'.format(exam_id=exam_id)
+                    )
+                    log.error(log_message)
+                    continue
+
+                exam_time = target_exam["time_limit_mins"]
+                added_time = round(exam_time * multiplier)
+                time_allowed = str(added_time)
+                for user_id in user_ids:
+                    try:
+                        add_allowance_for_user(exam_id, user_id,
+                                               ProctoredExamStudentAllowance.ADDITIONAL_TIME_GRANTED,
+                                               time_allowed)
+                    except ProctoredBaseException:
+                        log_message = (
+                            'Failed to add allowance for user_id={user_id} '
+                            'for exam_id={exam_id}'.format(
+                                user_id=user_id,
+                                exam_id=exam_id
+                            )
+                        )
+                        log.error(log_message)
+        else:
+            raise AllowanceValueNotAllowedException(err_msg)
```

Comment:
Please move this raise function to the beginning of the logic, instead of the `else` statement here.